src/main/java/com/minecolonies/apiimp/initializer/ModBuildingsInitializer.java

```diff
@@ -137,8 +140,8 @@ public static void init(final RegistryEvent.Register<BuildingEntry> event)
 
         ModBuildings.home = new BuildingEntry.Builder()
                               .setBuildingBlock(ModBlocks.blockHutHome)
-                              .setBuildingProducer(BuildingHome::new)
-                              .setBuildingViewProducer(() -> BuildingHome.View::new)
+                              .setBuildingProducer((colony, blockPos) -> new DefaultBuildingInstance(colony, blockPos, "citizen", ModBuildings.home))
```

Comment:
This really really really feels like a code smell in my opinion. The building instance really should not care what building it is representing, neither should the modules which are inside it.

Reply:
The building instance absolutely does need it. Necessary for the schematics etc